src/lib/fetchData.js

```diff
@@ -1,29 +1,13 @@
 import $ from 'jquery';
 
-const TIMEOUT = 5000;
+const TIMEOUT = 30000;
 
 export default function fetchData(_url, options={}) {
   let _type = options.type || 'GET';
-  return new Promise(function (resolve, reject) {
```

Comment:
I'm a little worried about these changes to the fetchData return value. Is it necessary?

fetchData is used by many parts of the existing code. They might rely on the promise to resolve to certain value as defined by the current fetchData. Changing to the $.ajax might break things.

I imagine we might want to move away from jquery and use the `fetch API` for ajax calls in future. Exposing the return value of $.ajax might lead to code relying on specific functions available only in $.ajax's return type, hence making it harder to migrate from.

Reply:
I totally understand what you're saying. 

The reason for this change is to support request cancellation. Fetching the expression annotations can be a little sluggish. If a user does a lot of filtering or paginating on the table there could be multiple requests in progress at the same time. Then there's a race condition with no guarantee of what order they'll come back. So I needed some way to cancel the in-progress request before the next one goes out. The `jqXHR` object that `.ajax` returns has a method to do that. It also is a Promise-like object so any `.then` and `.catch` calls that are currently chained to `fetchData` will continue to work.

I think we could benefit from request cancellation in other parts of the code. All the tables that do remote pagination could use it. The search autocomplete could really use it, too. But before we do that I agree we absolutely should convert from jquery to the Fetch API. Its cancellation mechanism seems to be a little more sophisticated, and I think it'll work well for us. So maybe rewriting `fetchData` to use the Fetch API is something we can do right at the start of the next release.